probes/internal/scripts/setup.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		panic("usage: make setup-probe probeName=arg")
+	}
+	pn := os.Args[1]
+
+	pd := path.Join("probes", pn)
+	err := os.Mkdir(pd, 0o700)
+	if err != nil {
+		panic(err)
+	}
+
+	y := time.Now().Year()
+
+	implBoilerplate := fmt.Sprintf(`// Copyright %d OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//nolint:stylecheck
+package %s
+
+import (
+	"embed"
+
+	"github.com/ossf/scorecard/v5/checker"
+	"github.com/ossf/scorecard/v5/finding"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const (
+	Probe = "%s"
+)
+
```

Comment:
There's registration code that can't be reflected here. Not that I'm sure it's something a template could help with, maybe have a placeholder comment describing the need to fill it out?